packages/statsd/src/client.ts

```diff
@@ -0,0 +1,125 @@
+import {StatsD, ClientOptions, Tags} from 'hot-shots';
+import {snakeCase} from 'change-case';
+
+export interface Timer {
+  stop(): number;
+}
+
+export interface Logger {
+  log(message: string): void;
+}
+
+export interface Options extends ClientOptions {
+  logger?: Logger;
+  snakeCase?: boolean;
+}
+
+export class StatsDClient {
+  private statsd: StatsD;
+  private logger: Logger | null = null;
+  private options: Options;
+
+  constructor({logger, ...options}: Options) {
+    this.logger = logger || null;
+    this.options = {
+      ...options,
+      errorHandler: options.errorHandler
+        ? options.errorHandler
+        : this.handleError.bind(this),
+    };
+
+    this.statsd = new StatsD(this.options);
+  }
+
+  distribution(stat: string | string[], value: number, tags?: Tags) {
+    return new Promise<void>(resolve => {
+      this.statsd.distribution(
+        stat,
+        value,
+        this.normalizeTags(tags),
+        this.createErrorCallback(resolve),
+      );
+    });
+  }
+
+  increment(stat: string | string[], tags?: Tags) {
+    return new Promise<void>(resolve => {
+      this.statsd.increment(
+        stat,
+        1,
+        this.normalizeTags(tags),
+        this.createErrorCallback(resolve),
+      );
+    });
+  }
+
+  close() {
+    return new Promise<void>(resolve => {
+      this.statsd.close(this.createErrorCallback(resolve));
+    });
+  }
+
+  addGlobalTags(globalTags: Tags) {
+    this.statsd = this.statsd.childClient({
+      globalTags,
+    });
+  }
+
+  initTimer(): Timer {
```

Comment:
I'm not super sure this makes sense as part of this class tbh, it seems unrelated to the rest, and doesn't share any state.

Reply:
good call, 🔥ing